course/src/main/java/org/openedx/course/presentation/calendarsync/CalendarSyncDialog.kt

```diff
@@ -0,0 +1,237 @@
+package org.openedx.course.presentation.calendarsync
+
+import android.content.res.Configuration
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.AlertDialog
+import androidx.compose.material.CircularProgressIndicator
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Dialog
+import org.openedx.core.presentation.global.app_upgrade.TransparentTextButton
+import org.openedx.core.ui.theme.OpenEdXTheme
+import org.openedx.core.ui.theme.appColors
+import org.openedx.core.ui.theme.appShapes
+import org.openedx.core.ui.theme.appTypography
+import org.openedx.course.R
+import androidx.compose.ui.window.DialogProperties as AlertDialogProperties
+
+@Composable
+fun CalendarSyncDialog(
+    syncDialogType: CalendarSyncDialogType,
+    calendarTitle: String,
+    syncDialogAction: (CalendarSyncDialogType) -> Unit,
+    dismissSyncDialog: () -> Unit,
+) {
+    when (syncDialogType) {
```

Comment:
IMO, we can optimize it more

Reply:
I tried multiple ways, though no doubt the code was optimize but the end result was really messy. That's why I kept it as it is.